src/clients/c++/request_common.cc

```diff
@@ -214,6 +216,17 @@ InputImpl::SetRaw(const std::vector<uint8_t>& input)
   return SetRaw(&input[0], input.size());
 }
 
+Error
+InputImpl::SetSharedMemory(
+    const std::string& name, size_t offset, size_t byte_size)
+{
+  shm_name_ = name;
```

Comment:
Check if io_type_ is RAW and return error.
Check if io_type_ is SHARED_MEMORY and return error that can only set once.